File: 12-VowelCount/12.js
function VowelCountRegex(str){
  var result = str.match(/[aeiou]/gi)
  return (result) ? result.length : 0
}

function VowelCount(str) { 
  return str.split('').reduce(function(acc,c){
      return acc + (/[aeiou]/gi.test(c) ? 1 : 0)
    },0)
}

console.log(VowelCount("helloo") === 3)
console.log(VowelCount("sdjk")   === 0)

File: 12-VowelCount/12.hs
--Count the number of vowels in a word
vowelCount :: String -> Int
vowelCount = foldr (\x acc -> if elem x "aeiou" then acc + 1 else acc) 0

main = do
  print $ vowelCount "helloo" == 3
  print $ vowelCount "ghjk"   == 0